chalice/app.py

```diff
@@ -495,6 +500,49 @@ def default_binary_types(self):
         return list(self._DEFAULT_BINARY_TYPES)
 
 
+class WebsocketAPI(object):
+    _WEBSOCKET_ENDPOINT_TEMPLATE = 'https://{domain_name}/{stage}'
+
+    def __init__(self):
+        self.session = None
+        self._client = None
+
+    def configure(self, domain_name, stage):
+        if self.session is None:
+            raise ValueError(
+                'Assign app.websocket_api.session to a boto3 session before '
+                'using the WebsocketAPI'
+            )
+        if self._client is not None:
+            return
+        endpoint = self._WEBSOCKET_ENDPOINT_TEMPLATE.format(
+            domain_name=domain_name,
+            stage=stage,
+        )
+        self._client = self.session.client(
```

Comment:
It'd be nice to defer this client creation until the `send()` call given the `client()` call can have side effects.